vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -5,6 +5,27 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
+// HealthConfig holds configuration settings for the HEALTHCHECK feature.
+type HealthConfig struct {
+	// Test to perform to check container is healthy.
+	// An empty slice means to inherit the default.
+	// Otherwise, Cmd[0] is "NONE" or "CMD".
+	Test strslice.StrSlice `json:",omitempty"`
+
+	// Nil means to inherit.
+	GracePeriod *float64 `json:",omitempty"` // Time after container start to assume failure means still starting.
```

Comment:
I'd really recommend avoiding trying to define an inheritance scheme for doing overrides. Just place these into a separate structure to allow overriding the parameters for the check or the just the test. Placing pointers across these fields is unnecessary. Look around in this file; they aren't there.

Looking back, I am trying to find the PR against engine-api for these changes. It looks like this needs to go for further review.